File: 0009-palindrome-number/0009-palindrome-number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        if(x<0){
            return false;
        }
        long long reverse = 0;
        int temp = x;
        while(temp){
            int digit = temp%10;
            temp = temp/10;
            reverse = reverse*10 + digit;
        }
        if(x==reverse){
            return true;
        } else {
            return false;
        }
    }
};

File: 0009-palindrome-number/0009-palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
        int y = x;
        int reversedX = 0;
        while(y!=0){
            int lastDigit = y%10;
            reversedX = reversedX*10 + lastDigit;
            y/=10;
        }
        return reversedX == x && x >= 0;
    }
}